pkg/controllers/provisioningv2/fleetcluster/fleetcluster.go

```diff
@@ -29,6 +30,11 @@ type handler struct {
 	apply         apply.Apply
 }
 
+// Register registers the fleetcluster controller, which is responsible for creating fleet cluster objects.
+// When fleet cluster objects are created, Fleet uses the object to deploy the fleet-agent into the cluster. Notably,
```

Comment:
I'm hoping this comment helps future engineers find the source of `fleet-agent` when searching the codebase for it.

Reply:
Thank you for adding that, it clears up a lot of the confusion I had initially